returnn/hdf.py

```diff
@@ -207,26 +207,38 @@ class BlissToPcmHDFJob(Job):
     compatible with the RETURNN HDFDataset
     """
 
+    __sis_hash_exclude__ = {"multi_channel_strategy": "none"}
+
     def __init__(
         self,
         bliss_corpus: tk.Path,
         segment_file: Optional[tk.Path] = None,
         output_dtype: str = "int16",
+        multi_channel_strategy: str = "none",
```

Comment:
Wouldn't it be nicer to have the actual `None` as default? Also, setting an integer to pick the channel seems quite straight-forward. More advanced things like channel-averaging might be specified via strings.

Reply:
> Wouldn't it be nicer to have the actual None as default? 

I was thinking this would be "an existing strategy to do mothing" instead of "no strategy", but I am open to have `None` as default (or change it to `"assert_single_channel"`).

> Also, setting an integer to pick the channel seems quite straight-forward. More advanced things like channel-averaging might be specified via strings.

I would like to avoid very elaborate multi-typing, e.g.
```
multi_channel_strategy: Optional[Union[str,int]] = None,
```

The "most correct" thing would be to have
```
class strategy(enum.Enum):
  ASSERT_SINGLE_CHANNEL = enum.auto()
  PICK_NTH = enum.auto()

[...]
        multi_channel_strategy: strategy = strategy.ASSERT_SINGLE_CHANNEL,
        multi_channel_options: Dict[str,Any] = None
```
where in the case `PICK_NTH` we can select the n in the options, and e.g. for averaging we can have arithmetic or geometric, or define some weights...

But idk, if you want to go as far now..

Reply:
Well, I don't find the multi-typing you described so terrible. But I'm fine with leaving it as a string as well. I agree we shouldn't overcomplicate as long as only one case is needed.

Reply:
The strategy can even be an object holding the options itself, I think I would prefer that solution. So it can be a dataclass inherited from an "empty" class instead of an enum.

so something like (defined maybe within the job):
```python
class Strategy():
    pass
    
 class AssertSingleChannel(Strategy):
     pass
 
 @dataclass(frozen=True)
 class PickNth(Strategy):
     channel: int
 ```

Reply:
I tried your approach now. I was unsure about how to have an abstract baseclass `Strategy` as a type that itself never should be given as an argument. (And because of the "else" both `Strategy` and `AssertSingleChannel` were redundant in all but name.)

So now instead the `BaseStrategy` means no handling i.e. assert single channel and the only derived strategy `PickNth` is a dataclass.